testtools/deferredruntest.py

```diff
@@ -232,8 +233,10 @@ def _run_core(self):
             [error_observer.gotEvent, full_observer.emit],
             self._blocking_run_deferred, spinner)
 
+        full_log.seek(0, os.SEEK_SET)
```

Comment:
Since we know full_log is a StringIO and since we're pulling the content out, calling full_log.getvalue() would be simpler and more efficient.